numpy/linalg/linalg.py

```diff
@@ -1142,6 +1144,8 @@ def eig(a):
     _assertNdSquareness(a)
```

Comment:
Eric, is my PR removing that weird error in some way useful here, or is it orthogonal. Frankly, I am not sure myself whether or not it helps here, considering that lapack probably cannot handle 0d anyway.